test/test_evaluate/test_docstring.py

```diff
@@ -150,3 +159,169 @@ def foobar(x, y):
+        script = jedi.Script(s)
+        func = script._get_module().names_dict['foobar'][0].parent
+        evaluator = script._evaluator
+        types = docstrings.find_return_types(evaluator, func)
+        assert len(types) == 1
+        assert types[0].base.obj is builtins.int
+
+    # ---- Google Style Tests ---
+
+    def test_googlestyle_docstring(self):
+        s = dedent('''
+        def foobar(x, y):
+            """
+            Args:
+                x (int):
+                y (str):
+            """
+            y.''')
+        names = [c.name for c in jedi.Script(s).completions()]
+        assert 'isupper' in names
+        assert 'capitalize' in names
+
+    def test_googledoc_docstring_set_of_values(self):
+        s = dedent('''
+        def foobar(x, y):
+            """
+            Args:
+                x ({'foo', 'bar', 100500}):
+            """
+            x.''')
+        names = [c.name for c in jedi.Script(s).completions()]
+        assert 'isupper' in names
+        assert 'capitalize' in names
+        assert 'numerator' in names
+
+    def test_googledoc_alternative_types(self):
+        s = dedent('''
+        def foobar(x, y):
+            """
+            Args:
+                x (int or str or list):
+            """
+            x.''')
+        names = [c.name for c in jedi.Script(s).completions()]
+        assert 'isupper' in names
+        assert 'capitalize' in names
+        assert 'numerator' in names
+        assert 'append' in names
+
+    def test_google_returns(self):
+        s = dedent('''
+        def foobar(x, y):
+            """
+            Returns:
+                int: sum of x and y
+            """
+            return x + y
+
+        def bazbiz():
+            z = foobar(2, 2)
+            z.''')
+        script = jedi.Script(s)
+        names = [c.name for c in script.completions()]
+        assert 'numerator' in names
+
+    def test_google_follow_args(self):
+        from jedi.evaluate import docstrings
+        from jedi._compatibility import builtins
+        google_source = dedent('''
+        def foobar(x, y):
+            """
+            Args:
+                x (int or str or list):
+                y ({'foo', 'bar', 100500}):
+            """
+            ''')
+        script = jedi.Script(google_source)
+        func = script._get_module().names_dict['foobar'][0].parent
+        evaluator = script._evaluator
+        x_param = func.names_dict['x'][0].parent
+        y_param = func.names_dict['y'][0].parent
+        x_type_list = docstrings.follow_param(evaluator, x_param)
+        y_type_list = docstrings.follow_param(evaluator, y_param)
+        assert len(x_type_list) == 3
+        assert len(y_type_list) == 2
+        y_base_objs = set([t.base.obj for t in y_type_list])
+        x_base_objs = set([t.base.obj for t in x_type_list])
+        assert x_base_objs == {builtins.int, builtins.str, builtins.list}
+        assert y_base_objs == {builtins.int, builtins.str}
```

Comment:
Not sure if you want the jedi.evaluate.docstrings functions explicitly tested here, but it helped me very much in writing this PR to have an example that gives me an entry point in the actual code.